scripts/infra/cloud-instance.sh

```diff
@@ -291,11 +293,12 @@ install_rh_nvidia_drivers() {
     local cloud_type=$1
     "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" "sudo dnf install kernel-devel-\$(uname -r) kernel-headers-\$(uname -r) -y"
     "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm -y
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf clean expire-cache
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf module install nvidia-driver:latest-dkms -y
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install cuda-toolkit nvidia-gds libnccl-devel libcudnn8 nvidia-driver-NVML nvidia-driver-cuda-libs -y
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf update -y
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install https://developer.download.nvidia.com/compute/cuda/12.4.0/local_installers/cuda-repo-rhel9-12-4-local-12.4.0_550.54.14-1.x86_64.rpm -y
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install cuda-12-4 nvidia-gds-12-4 -y
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo -y
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install libcudnn8 libnccl-2.21.5-1+cuda12.4 libnccl-devel-2.21.5-1+cuda12.4 libnccl-static-2.21.5-1+cuda12.4 -y
```

Comment:
in @fabiendupont 's images he was using cudnn 9 here. In my local testing I was doing the same thing you are doing here, just noting the delta.

Reply:
I'll try that on the next round.

Reply:
This is what I get with cudnn9:

Collecting vllm@ git+https://github.com/opendatahub-io/vllm@2024.06.25 (from -r requirements-vllm-cuda.txt (line 1))
  Cloning https://github.com/opendatahub-io/vllm (to revision 2024.06.25) to /tmp/pip-install-8c_pg9rl/vllm_9456235180e045b08f85705f957b6439
  Running command git clone --filter=blob:none --quiet https://github.com/opendatahub-io/vllm /tmp/pip-install-8c_pg9rl/vllm_9456235180e045b08f85705f957b6439
  Running command git checkout -q 7cc6a9b588a4b08d095e22c49aeaebffbd6e64f4
  Resolved https://github.com/opendatahub-io/vllm to commit 7cc6a9b588a4b08d095e22c49aeaebffbd6e64f4
  Installing build dependencies: started
  Installing build dependencies: still running...
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      Traceback (most recent call last):
        File "/home/ec2-user/instructlab/venv/lib64/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/ec2-user/instructlab/venv/lib64/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/ec2-user/instructlab/venv/lib64/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-cnv06gx_/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 327, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-cnv06gx_/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 297, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-cnv06gx_/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 313, in run_setup
          exec(code, locals())
        File "<string>", line 11, in <module>
        File "/tmp/pip-build-env-cnv06gx_/overlay/lib64/python3.11/site-packages/torch/__init__.py", line 237, in <module>
          from torch._C import *  # noqa: F403
          ^^^^^^^^^^^^^^^^^^^^^^
      ImportError: libcudnn.so.8: cannot open shared object file: No such file or directory
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.